### PR TITLE
III-4890 Refactor CultuurNet\UDB3\Http\Proxy\Proxy

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

- throw `ApiProblem` on `Proxy` Errors

### Changed

- Make factory methods in `Proxy` instead of using Child classes.
- Make Proxy implement `RequestHandlerInterface`
- Changed bindings for `Proxy`

### Removed

- `FilterPathMethodProxy`, `CdbXmlProxy` & tests

---
Ticket: https://jira.uitdatabank.be/browse/III-4890

### CODE REVIEW

File: web/index.php [hunk 1]
```diff
@@ -64,10 +65,14 @@
     $app['config']['cdbxml_proxy']['enabled']) {
     $app->before(
         function (Request $request, Application $app) {
-            /** @var \CultuurNet\UDB3\Http\Proxy\CdbXmlProxy $cdbXmlProxy */
+            $psr7Request = (new DiactorosFactory())->createRequest($request);
+            /** @var \CultuurNet\UDB3\Http\Proxy\Proxy $cdbXmlProxy */
             $cdbXmlProxy = $app['cdbxml_proxy'];
 
-            return $cdbXmlProxy->handle($request);
+            $psr7Response = $cdbXmlProxy->handle($psr7Request);
```
Comment (JonasVHG) [inline web/index.php:72]: I'm pretty sure this cdbXml part can be removed. But maybe this should be done in a separate PR?

**Comment by bertramakers** [comment]
(Removed myself as a reviewer for now so this doesn't pop up in my daily reminder of PR review requests. When you mark it as ready for review I'll automatically become assigned again)

**Comment by bertramakers** [comment]
Replaced by https://github.com/cultuurnet/udb3-silex/pull/1151
